configure.ac

```diff
@@ -104,6 +104,13 @@ AS_IF([test "$SYS" = linux],[
 ])
 AM_CONDITIONAL(HAVE_ANDROID, test "${HAVE_ANDROID}" = "1")
 
+HAVE_WINDOWS=0
+case "${host_os}" in
```

Comment:
perhaps this is a bit of a nit-pick, but this looks like it may break cross-compilation. i.e. the *host OS* may be different from the *target OS*. But I'm also quite ignorant of how autotools work, so maybe there's no hope to support cross compilation anyway.

Reply:
Actually, I'm trying to push this PR so as to cross compile for windows. ${host_os} references the system on which the executable will run, in contrast with ${build_os} which is the system on which it is build. I also checked before right here, because it was confusing: https://www.gnu.org/software/autoconf/manual/autoconf-2.65/html_node/Specifying-Target-Triplets.html

But you may be right in the case of a Canadian cross, and I know absolutely nothing about it. If you think it should be supported, I'll do some research and fix this.

Currently, I don't differentiate WIN32 from WIN64 too, I don't know if you want to as nothing depends on this in libtorrent.

Reply:
Also, the build with MinGW currently needs morganz/mingw-std-threads header files because there is no C++ thread feature support in the mingw compiler with win32 thread model. I tried to alias boost thread types but I can't alias std::chrono types as they are directly used in the code (and already defined). Are you using features that std has but boost hasn't ?

Reply:
I don't really have a strong opinion, if it's fine with you you can leave it as is.

I actually build mingw on appveyor without any additional stuff (as far as I know). But I only got it to build with mingw64.

Reply:
Would you accept a patch to use boost types (aliased in a header) if C++ thread library is not available and if it is possible to write ?

About this PR, I hope I'll manage to finish it until the end of the week. It doesn't seems correct to me yet.

Reply:
I would be a bit hesitant to fake C++11 headers. Are you suggesting they are not available for mingw? I believe there would be a somewhat high maintenance burden to keep it up. I believe boost threads isn't quite compatible with std threads (but I could be wrong, I haven't looked in a long time). the ``thread`` class itself is probably the simplest part to support, but ``atomic<>``, ``mutex<>``, ``lock_guard<>``, ``unique_lock<>``, ``this_thread::yield()`` and ``this_thread::slee_for()`` may be trickier. And to put them in the ``std`` namespace is technically not allowed.

Reply:
The C++11 thread library as a whole is not available for mingw64 with win32 thread model. There are header-only implementations of these (meganz/mingw-std-threads) but you have to include and distribute them with the library so I think it's also a no-go. 

I don't find any other clean solution so I believe it's better if I only keep a local patch for these headers.